src/app/services/myinfo/myinfo.service.ts

```diff
@@ -0,0 +1,315 @@
+        appId: myInfoPrefix + singpassEserviceId,
+        mode: myInfoClientMode,
+      })
+    } else {
+      myInfoGovClient = new MyInfoGovClient({
+        realm,
+        singpassEserviceId,
+        privateKey: fs.readFileSync(myInfoKeyPath),
+        appId: myInfoPrefix + singpassEserviceId,
+        mode: MyInfoClientMode.Dev,
+      })
+      myInfoGovClient.baseUrl = MYINFO_DEV_BASE_URL
+    }
+
+    this.#myInfoClientBreaker = new CircuitBreaker(
+      (params) => myInfoGovClient.getPersonBasic(params),
+      {
+        errorThresholdPercentage: 80, // % of errors before breaker trips
+        timeout: 5000, // max time before individual request fails, ms
+        rollingCountTimeout: 30000, // width of statistical window, ms
+        volumeThreshold: 5, // min number of requests within statistical window before breaker trips
+      },
+    )
+  }
+
+  /**
+   * Fetches MyInfo person detail with given params.
+   * This function has circuit breaking built into it, and will throw an error
+   * if any recent usages of this function returned an error.
+   * @param params The params required to retrieve the data.
+   * @param params.uinFin The uin/fin of the person's data to retrieve.
+   * @param params.requestedAttributes The requested attributes to fetch.
+   * @param params.singpassEserviceId The eservice id of the form requesting the data.
+   * @returns the person object retrieved.
+   * @throws an error on fetch failure or if circuit breaker is in the opened state. Use {@link CircuitBreaker#isOurError} to determine if a rejection was a result of the circuit breaker or the action.
+   */
+  fetchMyInfoPersonData(
+    params: IPersonBasicRequest,
+  ): ResultAsync<IPersonBasic, CircuitBreakerError | FetchMyInfoError> {
+    return ResultAsync.fromPromise(
+      this.#myInfoClientBreaker.fire(params),
+      (error) => {
+        const logMeta = {
+          action: 'fetchMyInfoPersonData',
+          requestedAttributes: params.requestedAttributes,
+          eServiceId: params.singpassEserviceId,
+        }
+        if (CircuitBreaker.isOurError(error)) {
+          logger.error({
+            message: 'Circuit breaker tripped',
+            meta: logMeta,
+            error,
+          })
+          return new CircuitBreakerError()
+        } else {
+          logger.error({
+            message: 'Error retrieving data from MyInfo',
+            meta: logMeta,
+            error,
+          })
+          return new FetchMyInfoError()
+        }
+      },
+    )
+  }
+
+  /**
+   * Prefill given current form fields with given MyInfo data.
+   * @param myInfoData
+   * @param currFormFields
+   * @returns currFormFields with the MyInfo fields prefilled with data from myInfoData
+   */
+  prefillMyInfoFields(
+    myInfoData: IPersonBasic,
+    currFormFields: IFieldSchema[],
+  ): Result<IPossiblyPrefilledField[], never> {
+    return ok(
+      currFormFields.map((field) => {
```

Comment:
nit: a little hard to parse. Can assign it to a variable first, then only return at the end?

```ts
const prefilledFields = .....

return ok(prefilledFields)
```